tests/kani/Projection/double_projection.rs

```diff
@@ -0,0 +1,48 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+// kani-flags: --unwind 3
+
+//! Check that nested fat pointers work. This used to trigger an issue.
+//! The projection should only keep track of the inner most dereferenced
+//! element.
```

Comment:
Not that I'm aware. It just makes sense because once you are past a projection, you cannot go back. I added a few more test cases that shows that.

Unfortunately, I found an issue with `size_of_val` while doing that.